HIP/hip-980.md

```diff
@@ -0,0 +1,114 @@
+---
+hip: 980
+title: Enhance mirror node rest APIs with timestamp query parameter
+author: Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>
+working-group: Steven Sheehy <@steven-sheehy>, Nana Essilfie-Conduah <nana@swirldslabs.com>
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Accepted
+created: 2024-06-03
+requested-by: Solo Labs
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/981
+updated: 2024-06-25
+---
+
+## Abstract
+
+This HIP enhances part of the mirror node REST APIs to support querying by timestamp. This will allow users to query for tokens, transactions, and accounts that were created at a specific timestamp.
+
+## Motivation
+
+To enhance user experience and allow users or developers to query data from a specific point in time we will add a timestamp query parameter to some of the REST APIs, which are missing this support.
+This change will be especially valuable for building the foundation of the so-called shadow fork functionality, where a user or developer can use a local mirror node instance but run it against a production environment state.
+Having this support, users or developers will have the ability to use public state data to simulate and troubleshoot existing issues or replay historical transactions using local mirror node instance against production state.
+
+## Rationale
+
+Users may want to run historical transactions on a local mirror node instance using public state data. To make this possible, some of the existing REST APIs should be enhanced with a timestamp query parameter, so that historical public data can be fetched and used
+for the simulations.
+
+## User stories
+
+- As a user of mirror node, I want to be able to query historical data for accounts, contracts, tokens, and transactions that were created at a specific timestamp.
+- As a user of mirror node, I want to be able to run locally an `eth_call`, `eth_estimateGas` or `eth_debugTraceTransaction` call that fails on a production environment against the corresponding public state.
+- As a developer of mirror node, I want to be able to test new feature locally against a real life data from a public environment, so that I have better understanding of the feature behavior.
```

Comment:
```suggestion
- As a mirror node developer, I want to be able to test new features locally against production state so that I have better understanding of the feature behavior.
```

Reply:
Fixed